cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java

```
@@ -0,0 +1,529 @@
+                if (auth.getUsername() != null && auth.getPasswordSecret() != null) {
+                    kafkaBridgeCluster.setUsernameAndPassword(auth.getUsername(), auth.getPasswordSecret());
+                    kafkaBridgeCluster.setSaslMechanism(auth.getType());
+                } else  {
+                    log.warn("PLAIN authentication selected, but no username and password configured.");
+                    throw new InvalidResourceException("PLAIN authentication selected, but no username and password configured.");
+                }
+            }
+        }
+
+        if (spec.getTemplate() != null) {
+            KafkaBridgeTemplate template = spec.getTemplate();
+
+            if (template.getDeployment() != null && template.getDeployment().getMetadata() != null)  {
+                kafkaBridgeCluster.templateDeploymentLabels = template.getDeployment().getMetadata().getLabels();
+                kafkaBridgeCluster.templateDeploymentAnnotations = template.getDeployment().getMetadata().getAnnotations();
+            }
+
+            ModelUtils.parsePodTemplate(kafkaBridgeCluster, template.getPod());
+
+            if (template.getApiService() != null && template.getApiService().getMetadata() != null)  {
+                kafkaBridgeCluster.templateServiceLabels = template.getApiService().getMetadata().getLabels();
+                kafkaBridgeCluster.templateServiceAnnotations = template.getApiService().getMetadata().getAnnotations();
+            }
+
+            ModelUtils.parsePodDisruptionBudgetTemplate(kafkaBridgeCluster, template.getPodDisruptionBudget());
+        }
+
+        if (spec.getHttp() != null) {
+            kafkaBridgeCluster.setHttpEnabled(true);
+            if (spec.getHttp().getPort() == HEALTH_CHECK_PORT) {
+                log.warn("HTTP port cannot be set to {}. This port is already used for heath check.", HEALTH_CHECK_PORT);
```

Since you don't have any solution here, I think you should do something like `throw new InvalidResourceException(...)`.